tests/components/xiaomi_miio/test_config_flow.py

```diff
@@ -218,3 +218,169 @@ async def test_zeroconf_missing_data(hass):
 
     assert result["type"] == "abort"
     assert result["reason"] == "not_xiaomi_miio"
+
+
+async def test_config_flow_step_device_connect_error(hass):
+    """Test config flow, device connection error."""
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {const.CONF_DEVICE: True},
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "device"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.xiaomi_miio.device.Device.info",
+        side_effect=DeviceException({}),
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: TEST_HOST, CONF_NAME: TEST_NAME, CONF_TOKEN: TEST_TOKEN},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "device"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_config_flow_step_unknown_device(hass):
+    """Test config flow, unknown device error."""
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {const.CONF_DEVICE: True},
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "device"
+    assert result["errors"] == {}
+
+    mock_info = get_mock_info(model="UNKNOWN")
+
+    with patch(
+        "homeassistant.components.xiaomi_miio.device.Device.info",
+        return_value=mock_info,
+    ), patch(
+        "homeassistant.components.xiaomi_miio.async_setup_entry", return_value=True
```

Comment:
We don't need to patch async_setup_entry since the result doesn't lead to create entry.

Reply:
Thanks, I removed it